iroh-resolver/src/unixfs.rs

```diff
@@ -98,21 +171,30 @@ impl UnixfsNode {
                     .cloned()
                     .ok_or_else(|| anyhow!("missing data"))?;
                 let inner = unixfs_pb::Data::decode(inner_data)?;
-                // ensure correct unixfs type
-                let _typ: DataType = inner.r#type.try_into()?;
+                let typ: DataType = inner.r#type.try_into()?;
+                let node = Node { outer, inner };
 
-                Ok(UnixfsNode::Pb { outer, inner })
+                // ensure correct unixfs type
+                match typ {
+                    DataType::Raw => todo!(),
+                    DataType::Directory => Ok(UnixfsNode::Directory(node)),
+                    DataType::File => Ok(UnixfsNode::File(node)),
+                    DataType::Symlink => Ok(UnixfsNode::Symlink(node)),
+                    DataType::HamtShard => Ok(UnixfsNode::HamtShard(node)),
+                    DataType::Metadata => bail!("unixfs metadata is not supported"),
```

Comment:
What is metadata (what does it carry)? Potentially a very useful bit in the future.

Reply:
it is not actually used by any impl

Reply:
got it, I'll keep this in mind for future reference